src/server/pkg/auth/handler.go

```diff
@@ -0,0 +1,73 @@
+package auth
+
+import (
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/auth"
+)
+
+type authHandler func(*client.APIClient, string) error
+
+// authDisabledOr wraps an authHandler and permits the RPC if authHandler succeeds or
+// if auth is disabled on the cluster
+func authDisabledOr(h authHandler) authHandler {
+	return func(pachClient *client.APIClient, fullMethod string) error {
+		err := h(pachClient, fullMethod)
+		if auth.IsErrNotActivated(err) {
+			return nil
+		}
+		return err
+	}
+}
+
+// authPartialAnd wraps an authHandler and permits the RPC if authHandler succeeds or
+// if auth is partially enabled on the cluster. Handlers only report partial activation
+// if they would otherwise succeed, so this
+func authPartialAnd(h authHandler) authHandler {
```

Comment:
One more naming idea: I think you could call this `ppsUserOr`.

Basically, the only way for `me.FullyActivated` in `authenticated()` or `admin()` to be false is if the PPS user is calling it during partial activation (otherwise `WhoAmI` returns `ErrNoToken`). AFAICT, if an interceptor doesn't use this, the PPS user is the thing being kept out.

`ppsUserOr(authenticated)` reads a little weirdly becuase the PPS user is authenticated, but maybe that's an implicit implementation detail of `authenticated`